clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TaskHealthCachingAgent.java

```diff
@@ -159,6 +167,7 @@ private TaskHealth inferHealthNetworkInterfacedContainer(AmazonElasticLoadBalanc
           .withTargets(
             new TargetDescription()
               .withId(networkInterface.getPrivateIpv4Address())
+              .withPort(taskDefinition.getContainerDefinitions().get(0).getPortMappings().get(0).getHostPort())
```

Comment:
Does this work with cluster (non-fargate) deployments?  I am genuinely unsure

Reply:
Agreed @clareliguori - it looks like container port is a required value, and host port is not. I figured I'd make that change today, and it should be safer. I do have concerns about the port mappings collection itself being marked as optional:

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdefinitions.html#cfn-ecs-taskdefinition-containerdefinition-portmappings

Was planning to play around with a test ECS classic cluster ASAP to get a better idea.